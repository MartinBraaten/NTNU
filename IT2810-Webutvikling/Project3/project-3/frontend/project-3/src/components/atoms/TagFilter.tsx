//Filter for filtering o champion tags
import {
    Autocomplete,
    TextField,
    ThemeProvider,
    Typography,
} from "@mui/material";
import Box from "@mui/material/Box";
import { useEffect, useState } from "react";
import { fieldTheme } from "../themes/themes";

//inspired by https://mui.com/material-ui/react-autocomplete/
//Options for champion tag
const championTags = [
    "Marksman",
    "Support",
    "Assassin",
    "Mage",
    "Fighter",
    "Tank",
];

export interface ITagFilter {
    change: boolean | undefined;
    setChange: (change: boolean) => void;
}

//Component for filtering champions on their tag
export default function TagFilter({ change, setChange }: ITagFilter) {
    const [value, setValue] = useState<string | null>(null);

    //Sets a state that runs a useEffect in HomePage that updates all filters when a filter is changed
    useEffect(() => {
        setChange(!change);
    }, [value]);
    return (
        <Box sx={{ width: "200px" }} role="TagFilter">
            <Typography
                align="center"
                variant="h6"
                sx={{ fontSize: "18px", fontWeight: "400" }}
            >
                Tag:
            </Typography>

            <ThemeProvider theme={fieldTheme}>
                <Autocomplete
                    value={value}
                    disablePortal
                    id="tagSelect"
                    options={championTags}
                    onChange={(event: any, tag: string | null) => {
                        setValue(tag);
                    }}
                    sx={{ width: "200px", color: "#fff" }}
                    renderInput={(params) => (
                        <TextField role="ChampionTag" {...params} label="Tag" />
                    )}
                />
            </ThemeProvider>
        </Box>
    );
}
