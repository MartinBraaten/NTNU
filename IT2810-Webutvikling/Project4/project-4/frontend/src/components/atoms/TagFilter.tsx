//Filter for filtering o champion tags
import {
    Autocomplete,
    TextField,
    ThemeProvider,
    Typography,
} from "@mui/material";
import Box from "@mui/material/Box";
import { fieldTheme } from "../themes/themes";
import { useDispatch, useSelector } from "react-redux";
import { selectFilters, setFilter } from "../store/FiltersReducer";

//inspired by https://mui.com/material-ui/react-autocomplete/
//Options for champion tag
const championTags = [
    "Marksman",
    "Support",
    "Assassin",
    "Mage",
    "Fighter",
    "Tank",
];

//Component for filtering champions on their tag
export default function TagFilter() {
    const dispatch = useDispatch();
    const filters = useSelector(selectFilters);

    return (
        <Box sx={{ width: "200px" }} role="TagFilter">
            <Typography
                align="center"
                variant="h6"
                sx={{ fontSize: "18px", fontWeight: "400" }}
            >
                Tag:
            </Typography>

            <ThemeProvider theme={fieldTheme}>
                <Autocomplete
                    value={filters.tag}
                    disablePortal
                    id="tagSelect"
                    options={championTags}
                    onChange={(event: any, tag: string | null) => {
                        dispatch(setFilter({ field: "tag", value: tag }));
                    }}
                    sx={{ width: "200px", color: "#fff" }}
                    renderInput={(params) => (
                        <TextField role="ChampionTag" {...params} label="Tag" />
                    )}
                />
            </ThemeProvider>
        </Box>
    );
}
