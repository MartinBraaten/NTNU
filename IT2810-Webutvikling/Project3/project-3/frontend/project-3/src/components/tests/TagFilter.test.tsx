/* eslint-disable testing-library/await-async-query */
/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { render } from "@testing-library/react";
// eslint-disable-next-line testing-library/no-dom-import
import { screen } from "@testing-library/dom";
import TagFilter from "../atoms/TagFilter";

test("Renders correctly", () => {
    function mockFunction(prop: boolean) { }
    let { container } = render(<TagFilter change={undefined} setChange={mockFunction}/>);
    expect(container).toMatchSnapshot();
});

test("Updates value on writing", () => {
    function mockFunction(prop: boolean) { }
    const { container } = render(
        <TagFilter change={undefined} setChange={mockFunction}/>
    );

    let typeComponent = screen.getByRole(/^ChampionTag/i);
    expect(typeComponent.innerHTML).toMatch("");


    typeComponent.innerHTML = "Tank";

    expect(typeComponent.innerHTML).toMatch("Tank");

});
