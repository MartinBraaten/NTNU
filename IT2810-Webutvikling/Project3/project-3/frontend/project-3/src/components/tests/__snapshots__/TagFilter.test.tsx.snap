// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1domaf0"
    role="TagFilter"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-fff52w-MuiTypography-root"
    >
      Tag:
    </h6>
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-9nom65-MuiAutocomplete-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        role="ChampionTag"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-15ng2gq-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Tag
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-8t02te-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-12qs7hp-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14dw23f-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-cu9mew-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Tag
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
